homeassistant/components/nobo_hub/__init__.py

```diff
@@ -0,0 +1,88 @@
+"""The Nobø Ecohub integration."""
+from __future__ import annotations
+
+import logging
+
+from pynobo import nobo
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_COMMAND_OFF, CONF_COMMAND_ON, CONF_IP_ADDRESS
+from homeassistant.core import HomeAssistant, callback
+
+from .const import CONF_SERIAL, DOMAIN, HUB, UNSUBSCRIBE
+
+PLATFORMS = ["climate"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Nobø Ecohub from a config entry."""
+
+    # As there currently is no way to import options from yaml
+    # when setting up a config entry, we fallback to adding
+    # the options to the config entry and pull them out here if
+    # they are missing from the options
+    _async_import_options_from_data_if_missing(hass, entry)
+
+    serial = entry.data.get(CONF_SERIAL)
+    ip_address = entry.data.get(CONF_IP_ADDRESS)
+    name = entry.title
+    discover = ip_address is None
+    hub = nobo(serial=serial, ip=ip_address, discover=discover, loop=hass.loop)
+
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = {HUB: hub}
```

Comment:
Since we don't need to track the unsub anymore, we can just store the hub directly.
```suggestion
    hass.data.setdefault(DOMAIN, {})
    hass.data[DOMAIN][entry.entry_id] = hub
```